File: lmax-disruptor/src/main/java/disruptor/LongConsumer.java
package disruptor;

import com.lmax.disruptor.EventHandler;
import java.util.function.Consumer;

public class LongConsumer implements EventHandler<Event> {

  private final Consumer<Long> consumeFunction;

  public LongConsumer(Consumer<Long> consumeFunction) {
    this.consumeFunction = consumeFunction;
  }

  @Override
  public void onEvent(Event event, long sequence, boolean endOfBatch) throws Exception {
    consumeFunction.accept(event.getData());
  }
}

File: kotlin-coroutines-channel/src/main/kotlin/trile/coroutines/LongConsumer.kt
package trile.coroutines

import java.util.function.Consumer
import kotlinx.coroutines.channels.Channel

class LongConsumer(
  private val channel: Channel<Long>,
  private val consumeFunction: Consumer<Long>
) {

  suspend fun consume(numberEvent: Long) {
    for (i in 1 until numberEvent + 1) {
      consumeFunction.accept(channel.receive())
    }
  }
}
